### PR TITLE
fix(drop-down): Calling open() only on a collapsed toggle #3810

### LINKED ISSUE #3810: Calling open() on an already opened IgxDropDown replays the opening animation
### Description  
The title describes the whole issue.

 * igniteui-angular version: 7.1.3
 * browser: N/A

### PR DESCRIPTION
**Pull request by kdinev**
Closes #3810 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them-

### CODE REVIEW

**Comment by bazal4o** [comment]
As discussed we need to return back the check into igxToggle and remove it from all the components which are using the toggle and make such check into the open/close methods.

File: projects/igniteui-angular/src/lib/directives/toggle/toggle.directive.ts [hunk 1]
```diff
@@ -173,6 +173,10 @@ export class IgxToggleDirective implements IToggleView, OnInit, OnDestroy {
      * ```
      */
     public open(overlaySettings?: OverlaySettings) {
+        if (!this._collapsed) {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/directives/toggle/toggle.directive.ts]: OK, question: isn't this required for the tooltip to actually replay its animation, and if so, wouldn't this break the tooltip behavior?

Comment (wnvko) [reply projects/igniteui-angular/src/lib/directives/toggle/toggle.directive.ts]: igxTooltip should work correctly now

File: projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts [hunk 2]
```diff
@@ -195,6 +198,12 @@ export class IgxTooltipTargetDirective extends IgxToggleActionDirective implemen
 
         this._overlayDefaults.positionStrategy = new AutoPositionStrategy(positionSettings);
         this._overlayDefaults.closeOnOutsideClick = false;
+        this.target.onClosed.subscribe(() => {
```
Comment (tachojelev) [inline projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts]: I believe we should unsubscribe from the tooltip's onClosed event when the respective targets gets destroyed.

File: projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts [hunk 3]
```diff
@@ -195,6 +198,12 @@ export class IgxTooltipTargetDirective extends IgxToggleActionDirective implemen
...
+            this.closeAnimationInProgress = false;
+        });
+        this.target.onOpened.subscribe(() => {
```
Comment (tachojelev) [inline projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts]: I believe we should unsubscribe from the tooltip's onOpened event when the respective targets gets destroyed.

Comment (wnvko) [reply projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts]: done

File: projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts (hunk 2 above)
Comment (wnvko) [reply projects/igniteui-angular/src/lib/directives/tooltip/tooltip.directive.ts]: done

File: projects/igniteui-angular/src/lib/services/overlay/overlay.ts [hunk 4]
```diff
@@ -508,12 +531,17 @@ export class IgxOverlayService implements OnDestroy {
         }
     }
 
-    private getOverlayById(id: string): OverlayInfo {
+    /** @hidden */
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/services/overlay/overlay.ts]: If this is intended to be internal, mark it `@internal`

Comment (wnvko) [reply projects/igniteui-angular/src/lib/services/overlay/overlay.ts]: added :)

**Pull request merged**